lib/screens/sub_category_list_view.dart

```diff
@@ -0,0 +1,112 @@
+
+class SubcategoryList extends StatefulWidget {
+  final String gamename;
+  final String gameid;
+
+  const SubcategoryList({Key key, this.gamename, this.gameid})
+      : super(key: key);
+  @override
+  MyappclassState createState() => new MyappclassState();
+}
+
+class MyappclassState extends State<SubcategoryList> {
+  @override
+  bool val = true;
+  String message = 'this is true';
+  List<CategoryTopic> _datacategory = new List<CategoryTopic>();
+  
+  var _categorydata;
+  @override
+  void initState() {
+    super.initState();
+    _initData();
+  }
+
+  void _initData() async {
+    String iddata = widget.gameid;
+    String idname = widget.gamename;
+    print(".....id matching or not.::$iddata......::$idname");
+    // var notifs = await NotifRepo().getNotifCountByType();
+    _categorydata =
+        await CategoryTopicRepo.categoryTopicDao.getAllCategoryTopic(iddata);
+    //  datatemplate= await  ActivityTemplateRepo.activityTemplateDao.getalltemplate();
+    print("object...category data is...eee...${_categorydata}");
+    setState(() {
+      _datacategory = _categorydata;
+      print(".......::database data is....${_datacategory.length}");
+      // _notifs = notifs;
+    });
+  }
+
+  Widget build(BuildContext context) {
+    MediaQueryData media = MediaQuery.of(context);
+    var size = media.size;
+    Orientation orientation = MediaQuery.of(context).orientation;
+    return new Scaffold(
+        appBar: new AppBar(
+          title: new Text("${widget.gamename}"),
+        ),
+        body: Stack(children: [
+          Container(
+              color: const Color(0xffFECE3D),
+              child: new GridView.count(
+                key: new Key('Game_page'),
+                primary: true,
+//          padding: const EdgeInsets.all(.0),
+                crossAxisSpacing: 12.0,
+                mainAxisSpacing: 12.0,
+                crossAxisCount: media.size.height > media.size.width ? 2 : 2,
+                children: new List.generate(_datacategory.length, (i) {
+                  return GestureDetector(
+                    onTap: () {
+                      String gamename = _categorydata[i].name;
+                      String gameid = _categorydata[i].id;
+                      Navigator.of(context).push(new MaterialPageRoute(
+                          builder: (BuildContext context) =>
+                              new SubcategoryList(
+                                  gamename: gamename, gameid: gameid)));
+                    },
+                    child: new Container(
+                      height: 40.0,
+                      width: 40.0,
+                      color: Colors.redAccent,
+                      child: Center(
+                          child: new Text("${_datacategory[i].topicId}")),
+                    ),
+                  );
+                }),
+              )),
+          new MyAppScrol(),
+        ]));
+  }
+}
+
+class MyAppScrol extends StatelessWidget {
```

Comment:
Please pay attention to class names. Ideally this should have been SubcategoryScroller or something like that